engine/runtime-integration-tests/src/test/java/org/enso/interpreter/test/semantic/DataflowErrorPropagationTest.java

```diff
@@ -0,0 +1,86 @@
+package org.enso.interpreter.test.semantic;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import org.enso.common.MethodNames;
+import org.enso.test.utils.ContextUtils;
+import org.graalvm.polyglot.Context;
+import org.graalvm.polyglot.PolyglotException;
+import org.graalvm.polyglot.Value;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class DataflowErrorPropagationTest {
+  private static Context ctx;
+  private static Value suppressError;
+  private static Value suppressErrorWithAssign;
+
+  @BeforeClass
+  public static void prepareCtx() {
+    ctx = ContextUtils.createDefaultContext();
+    var code =
+        """
+    from Standard.Base import all
+
+    private yield_error yes:Boolean -> Text =
+        if yes then Error.throw "Yielding an error" else
+            "OK"
+
+    suppress_error yes:Boolean value =
+        yield_error yes
```

Comment:
This 49c628ef1f06e4b4659512964e8c806fe9d9cf12 commit changes the behavior on this line:
- previously the `Error` returned from `yield_error` was ignore
- previously the execution continued to next line and `value` was returned
- now the `Error` is detected and execution of this method ends immediately with the `Error`

Reply:
Could you elaborate and show some example where you think the behaviour is not what we want?

From my perspective, with the changes from this PR all is good - the _data**flow**_ errors **flow** with the dataflow, determined by dependencies between method calls. The problem was when the value was unassigned because the dataflow error would be lost unhandled. But if it's assigned to something, it should flow further and be propagated. Unless the name is unused - but we have 'unused warning' for that already.

I think that propagating 'more aggressively' upon assignment would make the dataflow error not much (if at all in practice?) different from a Panic. Also it may be problematic for the IDE. The whole point of dataflow errors is that they _do not stop_ execution of other nodes if they are assigned to something, so that even if one node in the GUI errors, all the others can still be executed.

Reply:
> assignment to also propagate errors in this way? 
> This is another class of common missed errors, I would think.

Can such an error be missed without a 

> we have 'unused warning' for that already.

warning being printed by the compiler? If not, then it is less critical scenario. Users get compilation warning and that might warn them enough.

Btw. I think there is a contra-example:
```ruby
x = Error.throw "OK"
x . catch Any h->h
```
the second line wouldn't even be invoked.

Reply:
@radeusgd I definitely see your point -- assignments like this form the top-level spine of the entire program, so this would prevent execution after a dataflow error. For that case, I agree. For an assignment with a regular method call, one that does not represent a top-level node (from the user's point of view), it's still something that could be lost.

@JaroslavTulach I see your point here as well.  However, this case:

```
x = Error.throw "OK"
```

Will still mask errors, and hide them from the end user and the library developer. I don't have a suggestion for how to handle this, but it still should be handled. If something is called "error" then it should never be hidden.

Reply:
> mask errors, and hide them from ... the library developer

No, it won't hide them. There will be a warning about _unused variable `x`_. E.g. library developer has been notified the code is smelly.